bson/src/main/org/bson/codecs/pojo/PojoCodecImpl.java

```diff
@@ -266,9 +267,18 @@ private void decodeProperties(final BsonReader reader, final DecoderContext deco
         return codec;
     }
 
+    @SuppressWarnings("unchecked")
     private <S> Codec<S> getCodecFromPropertyRegistry(final PropertyModel<S> propertyModel) {
         try {
-            return propertyCodecRegistry.get(propertyModel.getTypeData());
+            Codec<S> codec = propertyCodecRegistry.get(propertyModel.getTypeData());
+            BsonType representation = propertyModel.getBsonRepresentation();
+            if (representation != null) {
+                if (codec instanceof RepresentationConfigurable) {
+                    return ((RepresentationConfigurable<S>) codec).withRepresentation(representation);
+                }
+                throw new IllegalArgumentException("Codec must implement RepresentationConfigurable to support BsonRepresentation");
```

Comment:
Why not CodecConfigurationException?

Reply:
This method is wrapped in a try-catch that catches CodecConfigurationExceptions. I can easily refactor the method, but I originally didn't think which error type we threw made a big difference, but your comment above made sense.

The catch is only really intended for that `propertyCodecRegistry.get` call anyway, so it's probably better to refactor anyway.

Reply:
Done